compiler/rustc_target/src/spec/apple_base.rs

```diff
@@ -221,6 +224,7 @@ fn link_env_remove(arch: Arch, os: &'static str) -> StaticCow<[StaticCow<str>]>
         // "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld",
         // although this is apparently ignored when using the linker at "/usr/bin/ld".
         env_remove.push("IPHONEOS_DEPLOYMENT_TARGET".into());
+        env_remove.push("TVOS_DEPLOYMENT_TARGET".into());
```

Comment:
Uh, preexisting so it doesn't have to be addressed here, but should we be doing this for "WATCHOS_DEPLOYMENT_TARGET"?

Reply:
I'll catch this after this PR merges. I think this mainly matters in cases involving build scripts that want to use SDK tools, but I could be mistaken.